### ISSUE TITLE
Union of legit types is also a legit type

### ISSUE THREAD

**Issue by sassanh** [microsoft/pyright#8390]
**Describe the bug**
In this code the last line causes a `reportArgumentType` pyright error.

**Code or Screenshots**
```python
class OriginalType: ...


class SubclassOne(OriginalType): ...


class SubclassTwo(OriginalType): ...


UnionType = SubclassOne | SubclassTwo


def some_function(_: type[OriginalType]): ...


some_function(OriginalType)
some_function(SubclassOne)
some_function(SubclassTwo)
some_function(UnionType) # <-- Pyright reports `UnionType` is not acceptable here
```

**Comment by erictraut** [comment]
This behavior is correct, so I don't consider this a bug. Pyright's behavior here is in conformance with the typing spec.

The type of `UnionType` when used in a value expression is `types.UnionType`.

```python
U = SubclassOne | SubclassTwo
reveal_type(U)  # UnionType
```

When it's used in a type expression, it acts as a type alias for the type `SubclassOne | SubclassTwo`.

In your case, you're using it in a value expression (an argument for a function call), and its type is not compatible with `type[OriginalType]`.

For comparison, all othe other major type checkers including mypy also generate an error here. This behavior is conformant with the typing spec. For details, refer to [this section](https://typing.readthedocs.io/en/latest/spec/special-types.html#type).

You may be interested in draft [PEP 747](https://peps.python.org/pep-0747/) that provides a way to annotate "type forms".
